packages/webapp/src/components/collection/deckPanel.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react'
+import { Deck, Card } from 'src/store/types' 
+import Image from 'next/image' 
+
+interface DeckConstructionPanelProps {
+    deck: Deck
+    selectedCards: Card[]
+    onCardSelect: (card: Card) => void
+    onSave: (deck: Deck) => void
+    onCancel: () => void
+  }
+  
+  
+  const DeckConstructionPanel : React.FC<DeckConstructionPanelProps> = ({ deck, selectedCards = [], onCardSelect, onSave, onCancel }) => {
+    const [ deckName, setDeckName ] = useState(deck.name)
+    const [ isDeckNameValid, setIsDeckNameValid ] = useState(true)
+
+    const handleDeckNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+      const newName = event.target.value
+      setDeckName(event.target.value)
+      setIsDeckNameValid(nameValid(newName))
+    }
+  
+    const nameValid = (name: string) => name.trim().length > 0
+
+    const handleSave = () => {
+      if (!nameValid(deckName)) {
+        setIsDeckNameValid(false)
+        return
+      }
+
+      const newDeck = {
+        name: deckName.trim(),
+        cards: selectedCards
+      }
+
+      onSave(newDeck)
+    }
+
+  return (
+    <div className="w-full flex flex-col items-center p-3">
+      <div className="flex justify-between items-center">
+        {/* Deck Name Input */}
+        <input
+          type="text"
+          value={deckName}
+          onChange={handleDeckNameChange}
+          style={{ outline: isDeckNameValid ? "none" : "2px solid red" }}
+          className="flex-grow px-2 py-2 border rounded-md text-gray-100 focus:outline-none focus:ring-2 focus:ring-blue-500 focus:border-transparent m-1.5"
+          placeholder=" Deck name"
+        />
+      
+        {/* Save and Cancel Buttons */}
+        <div className="flex justify-center">
+          <button onClick={handleSave} className="bg-green-500 hover:bg-green-600 text-white font-bold py-2 px-3 rounded-l-md">
+            ✓
+          </button>
+          <button onClick={onCancel} className="bg-yellow-500 hover:bg-yellow-600 text-white font-bold py-2 px-3 rounded-r-md">
+            ✕
+          </button>
+        </div>
+      </div>

```

Comment:
At normal size, this overflows on my Macbook (1728 x 1117)

<img width="282" alt="Screenshot 2024-02-16 at 17 05 05" src="https://github.com/0xFableOrg/0xFable/assets/202175/beff8861-a54f-4fe1-867e-c2f6a64904a2">

There should be no scrolling in this panel, the input field should resize to fit the width of the panel, and you should use flexbox so that the buttons flow underneath the input box when space is limited.

Reply:
I think we should also clarify this. Maybe move the buttons to the next line, replace ✓ with "Save deck" and ✕ with "Cancel".

Down the line, we'll gray out "save deck" if nothing changed, and pop a confirmation model if trying to cancel after making a change. But let's keep that for a further PR.

Reply:
The buttons are now on their own row and made explicit with "Save" and "Cancel".
Borders between cards are now removed.
Resizing looks good (to me) at least until it becomes very small.

![image](https://github.com/0xFableOrg/0xFable/assets/155228043/2feb54b8-c589-4bb5-b0ef-85aa744ee56a)
![image](https://github.com/0xFableOrg/0xFable/assets/155228043/2eee8ca2-fc1b-408c-894e-cfe7eade882b)

Is button auto scaling supported right now? It seems that the font currently dictates the size of them, even with w-auto (see `components/ui/button`).